### ISSUE TITLE
DQ Clustering

### ISSUE THREAD

**Issue by gh4nh** [sassoftware/sas-studio-custom-steps#154]
When entering a cluster rule in "Add rule 2" and uncheck the checkbox but keeping the columns in "Rule 2 columns", then they are still added to the dqmatch function. Idea to fix:

````
%macro usr_dqmatch() ;

	proc dqmatch data=&intable 
		out=&outtable
		cluster=&clusterName_name
		
		&blanks1
		%if &clusonly=1 %then %do ;
			CLUSTERS_ONLY
		%end ;
		;                                                                                                               

		%let con=1 ;
		%let addRule1=1;

		%do con=1 %to 5;
			%if &&addRule&con. %then %do;
				%do i = 1 %to &&rule&con._count;
					criteria condition=&con var=&&rule&con._&i._name exact;   
				%end;
			%end ;
		%end ;
	run;

%mend ;
%usr_dqmatch() ;
````

**Comment by snlwih** [comment]
@ArnoldToporowski, can you comment on this? Do you have time to fix this? If not, perhaps @gh4nh could create a pull request to fix this?

**Comment by snlwih** [comment]
Hiding a UI control doesn't clear the selections that were made in a control. This so that when the user unhides the control, their selections are still there. This is intended behaviour. Therefore when generating the criteria statement you need to check that the macro variable associated with the checkbox has a value of 1.

Although the code provided by @gh4nh works, I'd prefer it to read as follows which I think is easier to understand for less experienced SAS macro users:
```
%let addRule1=1; %* Rule 1 is required, so there is no checkbox for it in the UI;

%do con=1 %to 5;
    %if &&addRule&con=1 %then %do;
        %do i = 1 %to &&rule&con._count;
            criteria condition=&con var=&&rule&con._&i._name exact;   
        %end;
    %end ;
%end ;
```

**Comment by snlwih** [comment]
Fixed by #160
